src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -3227,3 +3187,15 @@ void pushNPCSpell(SPELL_TYPE spell) {
 void pushScrollSpell(SPELL_TYPE spell, unsigned int uPlayerID) {
     pushSpellOrRangedAttack(spell, uPlayerID, SCROLL_OR_NPC_SPELL_SKILL_VALUE, ON_CAST_CastViaScroll, 0);
 }
+
+void spellTargetPicked(int pid, int playerTarget) {
+    assert(pid != PID_INVALID || playerTarget != -1);
```

Comment:
I was under an impression that only one of these can be valid at a time, no?

So you'll need to xor them.